src/main/java/com/google/api/codegen/transformer/SampleTransformer.java

```diff
@@ -77,18 +76,60 @@ public SampleTransformer(SampleType sampleType) {
    * @param generator The function (typically a lambda) to generate the InitCode for a sample given
    *     an InitCodeContext
    * @param callingForms The list of calling forms applicable to this method, for which we will
-   *     generate samples if so configured via context.getMethodConfig()
+   *     generate samples if so configured via {@code context.getMethodConfig()}
+   */
+  public void generateSamples(
+      ApiMethodView.Builder methodViewBuilder,
+      MethodContext context,
+      Iterable<FieldConfig> fieldConfigs,
```

Comment:
Can we use `Collection` instead of `Iterable`? `Iterable` makes it really hard to use streaming API.
If it's too hard to do in this PR, I'm OK with it.

Reply:
I suggest we do this separately, in a different PR.